x/bank/internal/keeper/keeper.go

```diff
@@ -19,9 +19,6 @@ var _ Keeper = (*BaseKeeper)(nil)
 type Keeper interface {
 	SendKeeper
 
-	SetCoins(ctx sdk.Context, addr sdk.AccAddress, amt sdk.Coins) sdk.Error
-	SubtractCoins(ctx sdk.Context, addr sdk.AccAddress, amt sdk.Coins) (sdk.Coins, sdk.Error)
-	AddCoins(ctx sdk.Context, addr sdk.AccAddress, amt sdk.Coins) (sdk.Coins, sdk.Error)
 	InputOutputCoins(ctx sdk.Context, inputs []types.Input, outputs []types.Output) sdk.Error
```

Comment:
should we move this to the `SendKeeper`?

Reply:
I having a hard time recollecting why we did this...as in, what's the difference between `SendKeeper` and the entire `Keeper`?

But yes, let's move `InputOutputCoins` to the `SendKeeper` interface. Then `Keeper` will have delegation logic. Once/if that is removed, we can condense `Keeper` and `SendKeeper`.